### PR TITLE
Moved creation of http client outside of sendReport

### PR DESCRIPTION
**Pull request by bartsmykla**
Closes https://github.com/openfaas/openfaas-cloud/issues/354

Signed-off-by: Bart Smykla <bsmykla@vmware.com>

## Description
http.Client is created in root scope, so it's not recreated
every time, when sendReport is being invoked.

## How Has This Been Tested?
Deployed to my cluster on AWS.

## How are existing users impacted? What migration steps/scripts do we need?
They shouldn't be because the result should be the same, just performance should be improved.

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: gitlab-status/handler.go [hunk 1]
```diff
@@ -88,8 +90,8 @@ func sendReport(URL string, token string, state string, desc string, context str
 		return fmt.Errorf("error while creating request to GitLab API: %s", reqErr.Error())
 	}
 	req.Header.Set("PRIVATE-TOKEN", token)
-	client := &http.Client{}
```
Comment (alexellis) [inline gitlab-status/handler.go]: Can you use `http.DefaultClient` instead?

Comment (bartsmykla) [reply gitlab-status/handler.go]: Done

**Comment by bartsmykla** [comment]
Done

File: gitlab-status/handler.go [hunk 2]
```diff
@@ -77,6 +77,8 @@ func appendParameters(URL string, state string, desc string, context string) (st
 
 }
 
+var httpClient = http.DefaultClient
```
Comment (alexellis) [inline gitlab-status/handler.go]: No need for this pointer. Please just inline it.

Comment (bartsmykla) [reply gitlab-status/handler.go]: Done

**Pull request merged**